server/src/main/java/org/elasticsearch/action/support/local/TransportLocalClusterStateAction.java

```diff
@@ -125,4 +129,18 @@ public void onTimeout(TimeValue timeout) {
     private boolean isTaskCancelled(Task task) {
         return task instanceof CancellableTask cancellableTask && cancellableTask.isCancelled();
     }
+
+    // Remove the BWC support for the deprecated ?local parameter.
+    // NOTE: ensure each usage of this method has been deprecated for long enough to remove it.
+    @UpdateForV10(owner = UpdateForV10.Owner.DISTRIBUTED_COORDINATION)
+    public static void consumeDeprecatedLocalParameter(ToXContent.Params params) {
+        if (params.paramAsBoolean("local", false)) {
+            DeprecationLogger.getLogger(TransportLocalClusterStateAction.class)
+                .critical(
+                    DeprecationCategory.API,
+                    "TransportLocalClusterStateAction-local-parameter",
+                    "the [?local] query parameter to this API has no effect, is now deprecated, and will be removed in a future version"
+                );
+        }
```

Comment:
Do we need something to handle this with the rest compatibility header, so that if someone is using the 8.x compat version we don't emit this warning?

Reply:
Yeah, good catch. I thought about this before and I was on the fence about a good approach. Ideally, I'd like to do it in this method, so we only need to do the check in one place. However, then I'd have to change the type of the `params` parameter to `RestRequest` but that feels a little out of place in a transport action. Alternatively, I could move the method to a different class, but I didn't have great inspiration. Do you have any suggestions?

Reply:
I think perhaps `RestUtils`, and putting the consumption on the the `Rest*` class side (that passes the `RestRequest`) instead of in the Transport action. I don't think we want internal uses of the classes that pass `local` to emit deprecation warnings, because the user cannot do anything about those, so only on the REST side makes sense to me.

Reply:
Done in [57ea30e](https://github.com/elastic/elasticsearch/pull/116868/commits/57ea30e6817250edca71260fd62f743b8a9c57d2).